// @ionic-native/sqlite/index.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Injectable } from '@angular/core';
import { Cordova, CordovaInstance, Plugin, CordovaCheck, InstanceProperty } from '@ionic-native/core';
/**
 * @hidden
 */
export var SQLiteObject = (function () {
    function SQLiteObject(_objectInstance) {
        this._objectInstance = _objectInstance;
    }
    SQLiteObject.prototype.addTransaction = function (transaction) { };
    /**
     * @param fn {any}
     * @returns {Promise<any>}
     */
    SQLiteObject.prototype.transaction = function (fn) { return; };
    /**
     * @param fn {any}
     * @returns {Promise<any>}
     */
    SQLiteObject.prototype.readTransaction = function (fn) { return; };
    SQLiteObject.prototype.startNextTransaction = function () { };
    /**
     * @returns {Promise<any>}
     */
    SQLiteObject.prototype.close = function () { return; };
    SQLiteObject.prototype.start = function () { };
    /**
     * Execute SQL on the opened database. Note, you must call `openDatabase` first, and
     * ensure it resolved and successfully opened the database.
     */
    SQLiteObject.prototype.executeSql = function (statement, params) { return; };
    /**
     * @param sql
     * @param values
     * @returns {Promise<any>}
     */
    SQLiteObject.prototype.addStatement = function (sql, values) { return; };
    /**
     * @param sqlStatements {any}
     * @returns {Promise<any>}
     */
    SQLiteObject.prototype.sqlBatch = function (sqlStatements) { return; };
    SQLiteObject.prototype.abortallPendingTransactions = function () { };
    /**
     @param handler
     @param response
     */
    SQLiteObject.prototype.handleStatementSuccess = function (handler, response) { };
    /**
     * @param handler
     * @param response
     */
    SQLiteObject.prototype.handleStatementFailure = function (handler, response) { };
    SQLiteObject.prototype.run = function () { };
    /**
     * @param txFailure
     */
    SQLiteObject.prototype.abort = function (txFailure) { };
    SQLiteObject.prototype.finish = function () { };
    /**
     * @param sqlerror
     */
    SQLiteObject.prototype.abortFromQ = function (sqlerror) { };
    /**
     * @returns {Promise<any>}
     */
    SQLiteObject.prototype.echoTest = function () { return; };
    /**
     * @param first
     * @returns {Promise<any>}
     */
    SQLiteObject.prototype.deleteDatabase = function (first) { return; };
    __decorate([
        InstanceProperty, 
        __metadata('design:type', Object)
    ], SQLiteObject.prototype, "databaseFeatures", void 0);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object]), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "addTransaction", null);
    __decorate([
        CordovaInstance({
            successIndex: 2,
            errorIndex: 1
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object]), 
        __metadata('design:returntype', Promise)
    ], SQLiteObject.prototype, "transaction", null);
    __decorate([
        CordovaInstance(), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object]), 
        __metadata('design:returntype', Promise)
    ], SQLiteObject.prototype, "readTransaction", null);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', []), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "startNextTransaction", null);
    __decorate([
        CordovaInstance(), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', []), 
        __metadata('design:returntype', Promise)
    ], SQLiteObject.prototype, "close", null);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', []), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "start", null);
    __decorate([
        CordovaInstance(), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [String, Object]), 
        __metadata('design:returntype', Promise)
    ], SQLiteObject.prototype, "executeSql", null);
    __decorate([
        CordovaInstance(), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object, Object]), 
        __metadata('design:returntype', Promise)
    ], SQLiteObject.prototype, "addStatement", null);
    __decorate([
        CordovaInstance(), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object]), 
        __metadata('design:returntype', Promise)
    ], SQLiteObject.prototype, "sqlBatch", null);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', []), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "abortallPendingTransactions", null);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object, Object]), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "handleStatementSuccess", null);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object, Object]), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "handleStatementFailure", null);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', []), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "run", null);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object]), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "abort", null);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', []), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "finish", null);
    __decorate([
        CordovaInstance({
            sync: true
        }), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object]), 
        __metadata('design:returntype', void 0)
    ], SQLiteObject.prototype, "abortFromQ", null);
    __decorate([
        Cordova(), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', []), 
        __metadata('design:returntype', Promise)
    ], SQLiteObject.prototype, "echoTest", null);
    __decorate([
        Cordova(), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object]), 
        __metadata('design:returntype', Promise)
    ], SQLiteObject.prototype, "deleteDatabase", null);
    return SQLiteObject;
}());
/**
 * @name SQLite
 *
 * @description
 * Access SQLite databases on the device.
 *
 * @usage
 *
 * ```typescript
 * import { SQLite, SQLiteObject } from '@ionic-native/sqlite';
 *
 * constructor(private sqlite: SQLite) { }
 *
 * ...
 *
 * this.sqlite.create({
 *   name: 'data.db',
 *   location: 'default'
 * })
 *   .then((db: SQLiteObject) => {
 *
 *
 *     db.executeSql('create table danceMoves(name VARCHAR(32))', {})
 *       .then(() => console.log('Executed SQL'))
 *       .catch(e => console.log(e));
 *
 *
 *   })
 *   .catch(e => console.log(e));
 *
 * ```
 *
 * @classes
 * SQLiteObject
 */
export var SQLite = (function () {
    function SQLite() {
    }
    /**
     * Open or create a SQLite database file.
     *
     * See the plugin docs for an explanation of all options: https://github.com/litehelpers/Cordova-sqlite-storage#opening-a-database
     *
     * @param config the config for opening the database.
     * @return Promise<SQLiteObject>
     */
    SQLite.prototype.create = function (config) {
        return new Promise(function (resolve, reject) {
            sqlitePlugin.openDatabase(config, function (db) { return resolve(new SQLiteObject(db)); }, reject);
        });
    };
    SQLite.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    SQLite.ctorParameters = function () { return []; };
    __decorate([
        CordovaCheck(), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object]), 
        __metadata('design:returntype', Promise)
    ], SQLite.prototype, "create", null);
    SQLite = __decorate([
        Plugin({
            pluginName: 'SQLite',
            pluginRef: 'sqlitePlugin',
            plugin: 'cordova-sqlite-storage',
            repo: 'https://github.com/litehelpers/Cordova-sqlite-storage'
        }), 
        __metadata('design:paramtypes', [])
    ], SQLite);
    return SQLite;
}());
